job-view.php

```diff
@@ -16,20 +8,31 @@
 if (mysqli_num_rows($result) < 1) {
     header('Location: ' . BASE_URL);
 }
+$row = mysqli_fetch_assoc($result);
+
+
+//params for include template
+$meta_title = $row['title'] . " | Jobrix.tk";
 
-while ($row = mysqli_fetch_assoc($result)) :
-    ?>
-    <section class="section-fullwidth">
-        <div class="row">
-            <div class="job-single">
-                <div class="job-main">
-                    <div class="job-card">
-                        <div class="job-primary">
-                            <header class="job-header">
-                                <h2 class="job-title"><?php echo $row['title'] ?></h2>
-                                <div class="job-meta">
-                                    <a class="meta-company" href="#"><?php echo $row['company_name'] ?></a>
-                                    <span class="meta-date">
+//header template include
+require_once(dirname(__FILE__) . '/includes/theme-compat/header.php');
+
+//Checks if a number is passed as a parameter in the _get request and if the given ID exists in the database
+$jobs_id = "";
+if (!isset($_GET['id']) || !is_numeric($_GET['id'])) {
+    header('Location: ' . BASE_URL);
+} ?>
+<section class="section-fullwidth">
+    <div class="row">
+        <div class="job-single">
+            <div class="job-main">
+                <div class="job-card">
+                    <div class="job-primary">
+                        <header class="job-header">
+                            <h2 class="job-title"><?php echo $row['title'] ?></h2>
```

Comment:
we are displaying the `$row` items relying that we are going to have these fields for each job? the comments goes for al `$row` variable

Reply:
I created an empty array for all the elements that will be displayed.
Then for each I set the value from the DB if one exists.
This prevents an error from being thrown if an element is missing


$job = array(
	'title'               => '',
	'company_name'        => '',
	'company_description' => '',
	'company_site'        => '',
	'publication_date'    => '',
	'jobs_location'       => '',
	'salary'              => '',
	'description'         => '',
	'id_user'             => '',
);

foreach ( $row as $key => $value ) {
	if ( ! empty( $value ) && array_key_exists( $key, $job ) ) {
		$job[ $key ] = $value;
	}
}